diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp

```diff
@@ -370,11 +370,33 @@ class Updater : public DiagnosticTaskVector
    * \param h Node handle from which to get the diagnostic_period
    * parameter.
    */
-  explicit Updater(rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("diagnostics_updater"))
-  : node_(node), node_name_(node_->get_name())
+  template<class NodeT>
+  explicit Updater(std::shared_ptr<NodeT> node)
```

Comment:
It should be worth noting that you can't use this overload to initialize an `Updater` member instance from a class inheriting from NodeT. It appears that you can't use `shared_from_this()` from within a class' constructor, and since this interface can't take a raw pointer, I was forced to use the constructor which takes all four interfaces.

Not a huge deal since there is a workaround, but also not very convenient.

Reply:
should be addressed in https://github.com/ros/diagnostics/pull/112/commits/bf7aa601f056200963c174ddff74c6204c57585f

Reply:
Looks great, thanks!